components/brave_ads/core/internal/ads/serving/choose/eligible_ads_predictor_util.h

```diff
@@ -159,6 +165,44 @@ CreativeAdPredictorMap<T> ComputePredictorFeaturesAndScores(
   return creative_ad_predictors_with_features;
 }
 
+template <typename T>
+std::vector<int> ComputeVoteRegistry(
+    const std::vector<T>& creative_ads,
+    const TextEmbeddingHtmlEventList& text_embedding_html_events) {
+  DCHECK(!creative_ads.empty());
+
+  std::vector<int> vote_registry(creative_ads.size());
+
+  for (const auto& text_embedding_html_event : text_embedding_html_events) {
+    std::vector<float> similarity_scores;
+
+    for (const auto& creative_ad : creative_ads) {
+      const ml::VectorData ad_embedding(creative_ad.embedding);
+      const ml::VectorData page_text_embedding(
+          text_embedding_html_event.embedding);
+      const float similarity_score =
+          ad_embedding.ComputeSimilarity(page_text_embedding);
+
+      similarity_scores.push_back(similarity_score);
+    }
+
+    auto iter = base::ranges::max_element(
+        similarity_scores.cbegin(), similarity_scores.cend(),
+        [](const auto& lhs, const auto& rhs) { return lhs < rhs; });
+
+    while (iter != similarity_scores.end()) {
+      const size_t index = std::distance(similarity_scores.cbegin(), iter);
+      vote_registry[index]++;
+
+      iter =
```

Comment:
We are comparing floating point values here. I think we need to do a proper comparison here with something like:

```
base::IsApproximatelyEqual(f1, f2, std::numeric_limits<float>::epsilon())
```

Reply:
isn't this handled automatically by the `ranges::find`?

Reply:
unfortunately, no :-(